pydarn/plotting/fan.py

```diff
@@ -0,0 +1,248 @@
+
+    @classmethod
+    def plot_radar_fov(cls, dmap_data: List[dict], scan_index: int = 0,
+                       all_gates: bool = False, parameter: str = 'v',
+                       hemisphere: str = 'north', lowlat: int = 50,
+                       cmap: str = None, groundscatter: bool = False,
+                       zmin: int = None, zmax: int = None,
+                       colorbar: plt.colorbar = None,
+                       colorbar_label: str = ''):
+        # TODO: pyDARN can work out the hemisphere via superDARNRadars.radars
+        # check out the file. Even has an enum for hemisphere and associated
+        # value you will find in MAP files
+        """
+        Plots a radar FOV fan plot for the given data and scan number
+
+        Parameters
+        -----------
+            dmap_data: List[dict]
+                Named list of dictionaries obtained from SDarn_read
+            scan_index: int
+                Scan number from beginning of first record in file
+                Default: 0
+            parameter: str
+                Key name indicating which parameter to plot.
+                Default: v (Velocity). Alternatives: 'p_l', 'w_l', 'elv'
+            hemisphere: str
+                North or south, depending on radar. Currently,
+                pyDARN cannot work this out
+                itself, so you should always give it
+                Default: 'north'. Alternatives: 'south'
+            lowlat: int
+                Lower AACGM latitude boundary for the polar plot
+                Default: 50
+            all_gates: bool
+                Set to true to show all gates from a radar
+                Default: False(=75)
+            cmap: matplotlib.cm
+                matplotlib colour map
+                https://matplotlib.org/tutorials/colors/colormaps.html
+                Default: Official pyDARN colour map for given parameter
+            groundscatter : bool
+                Set true to indicate if groundscatter should be plotted in grey
+                Default: False
+            zmin: int
+                Minimum value for colouring
+                Default: {'p_l': [0], 'v': [-200], 'w_l': [0], 'elv': [0]}
+            zmax: int
+                Maximum value for colouring
+                Default: {'p_l': [50], 'v': [200], 'w_l': [250], 'elv': [50]}
+            colorbar: int
+                Draw a colourbar if set to an integer
+                Default: None
+            colorbar_label: str
+                the label that appears next to the colour bar
+                Default: ''
+        """
+        my_path = os.path.abspath(os.path.dirname(__file__))
+        base_path = os.path.join(my_path, '..')
+
+        # Setup scans for easy locating
+        scan_mark = [sub['scan'] for sub in dmap_data]
+        no_scans = 0
+        # TODO: my PEP8 linter says this variable is not used?
+        scan_no = 0
+        beam_scan = np.zeros((len(dmap_data)))
+        for beam in range(len(dmap_data)):
```

Comment:
Add a comment on what you are doing here please :)

Reply:
Resolved